### PR TITLE
Character and party SQL

### PR DESCRIPTION
**Pull request by mwitte98**

### CODE REVIEW

File: src/main/java/rpgcharacters/userflow/CharacterMenu.java [hunk 1]
```diff
@@ -152,6 +195,7 @@ private void printOptions () {
     * Defines the loop for this menu
     */
     public void enter ( Connection conn ) {
+        this.conn = conn;
```
Comment (returningsam) [inline src/main/java/rpgcharacters/userflow/CharacterMenu.java]: Maybe do this in the constructor?

**Comment by mwitte98** [comment]
Added SQL for creating and listing parties. Connection moved to constructor (this may cause merge conflicts with other PRs). Consistency with spaces

File: src/main/java/rpgcharacters/userflow/CreateCharacterMenu.java [hunk 2]
```diff
@@ -68,40 +91,60 @@ private int getStat(String statName, int curStatVal) {
...
-
-        curMaxAdd = 20;
-        curStats[0] = getStat("power",curStats[0]);
-        curStats[1] = getStat("proficiency",curStats[1]);
-        curStats[2] = getStat("personality",curStats[2]);
-        curStats[3] = getStat("perception",curStats[3]);
-
-        return saveToDB(charName,charStory,charRace,charArch,curStats);
+        try {
+            System.out.print("Character Race: ");
```
Comment (ekliot) [inline src/main/java/rpgcharacters/userflow/CreateCharacterMenu.java]: How does the user know what races are available? Replace this with a similar options menu to every other user selection (i.e. "1. Human")

File: src/main/java/rpgcharacters/userflow/CreateCharacterMenu.java [hunk 3]
```diff
@@ -68,40 +91,60 @@ private int getStat(String statName, int curStatVal) {
...
+            String charRace = sc.nextLine();
+            ResultSet raceResults = getStatRow("race", charRace);
+            if (raceResults == null) return false;
+
+            System.out.print("Character Archetype: ");
```
Comment (ekliot) [inline src/main/java/rpgcharacters/userflow/CreateCharacterMenu.java]: as with race

File: src/main/java/rpgcharacters/userflow/CreateCharacterMenu.java [hunk 4]
```diff
@@ -68,40 +91,60 @@ private int getStat(String statName, int curStatVal) {
...
+            String charArch = sc.nextLine();
+            ResultSet archResults = getStatRow("archetype", charArch);
+            if (archResults == null) return false;
+
+            int[] curStats = getCurStats(raceResults,archResults);
+
+            curMaxAdd = 20;
```
Comment (ekliot) [inline src/main/java/rpgcharacters/userflow/CreateCharacterMenu.java]: what is this used for?

File: src/main/java/rpgcharacters/userflow/CharacterMenu.java [hunk 5]
```diff
@@ -53,88 +70,114 @@ private String printChars () {
...
+                "-------------------------------------------------------\n" + // 50 chars
+                "User:        " + username + "\n" +
+                "Story:       " + story + "\n" +
+                "Race:        " + race + "\n" +
+                "Archetype:   " + arch + "\n" +
+                (party != null? "Party:       " + party + "\n" : "") +
+                "Power:       " + power + "\n" +
+                "Proficiency: " + proficiency + "\n" +
+                "Personality: " + personality + "\n" +
+                "Perception:  " + perception + "\n" +
```
Comment (ekliot) [inline src/main/java/rpgcharacters/userflow/CharacterMenu.java]: stats printing could be consolidated to two lines

```
System.out.println( "  POWER : PROFICIENCY : PERSONALITY : PERCEPTION" );
String modFormat =  "   %3d  :     %3d     :     %3d     :    %3d    ";
System.out.println( String.format( modFormat,
                      mods[0],   mods[1],      mods[2],     mods[3] )
);
```

Comment (mwitte98) [inline src/main/java/rpgcharacters/userflow/CharacterMenu.java]: I think we should leave it as is. It looks weird having them in all caps and experience by itself below.

File: src/main/java/rpgcharacters/userflow/CreateCharacterMenu.java (hunk 2 above)
Comment (mwitte98) [inline src/main/java/rpgcharacters/userflow/CreateCharacterMenu.java]: I agree and this was my first thought as well but I'll create a new issue for it and work on it later if there's time

File: src/main/java/rpgcharacters/userflow/CreateCharacterMenu.java (hunk 4 above)
Comment (mwitte98) [inline src/main/java/rpgcharacters/userflow/CreateCharacterMenu.java]: A user's custom mods can't be more than 20

**Pull request merged**